diff --git a/src/prime_factors/primes.py b/src/prime_factors/primes.py
index e69de29..4b825dc 100644
--- a/src/prime_factors/primes.py
+++ b/src/prime_factors/primes.py
@@ -1,2 +1,12 @@
 def generate(n: int):
-    return []
+    i = 2
+    factors = []
+    while i * i <= n:
+        if n % i:
+            i += 1
+        else:
+            n //= i
+            factors.append(i)
+    if n > 1:
+        factors.append(n)
+    return factors
